docs/docsite/rst/user_guide/playbooks_tags.rst

```diff
@@ -1,227 +1,402 @@
+
+  #. Add the same tag or tags to all tasks in the role by setting tags under ``roles``. See examples in this section.
+  #. Add the same tag or tags to all tasks in the role by setting tags on a static ``import_role`` in your playbook. See examples in :ref:`tags_on_imports`.
+  #. Add a tag or tags to to individual tasks or blocks within the role itself. This is the only approach that allows you to select or skip some tasks within the role. To select or skip tasks within the role, you must have tags set on individual tasks or blocks, use the dynamic ``include_role`` in your playbook, and add the same tag or tags to the include. When you use this approach, and then run your playbook with ``-tags foo``, Ansible runs the include itself plus any tasks in the role that also have the tag ``foo``. See :ref:`tags_on_includes` for details.
+
+When you incorporate a role in your playbook statically with the ``roles`` keyword, Ansible adds any tags you define to all the tasks in the role. For example:
+
+.. code-block:: yaml
+
+   roles:
+     - role: webserver
+       vars:
+         port: 5000
+       tags: [ web, foo ]
+
+or::
+
+   ---
+   - hosts: webservers
+     roles:
+       - role: foo
+         tags:
+           - bar
+           - baz
+       # using YAML shorthand, this is equivalent to:
+       # - { role: foo, tags: ["bar", "baz"] }
+
+.. _tags_on_imports:
+
+Adding tags to imports
+^^^^^^^^^^^^^^^^^^^^^^
+
+You can also apply a tag or tags to all the tasks imported by the static ``import_role`` and ``import_tasks`` statements:
+
+.. code-block:: yaml
+
+   ---
+   - hosts: webservers
+     tasks:
+       - import_role:
+           name: foo
+         tags:
+           - bar
+           - baz
+
+       - import_tasks: foo.yml
+         tags: [ web, foo ]
+
+.. _apply_keyword:
+
+Tag inheritance for includes: blocks and the ``apply`` keyword
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+By default, Ansible does not apply :ref:`tag inheritance <tag_inheritance>` to dynamic re-use with ``include_role`` and ``include_tasks``. If you add tags to an include, they apply only to the include itself, not to any tasks in the included file or role. This allows you to execute selected tasks within a role or task file - see :ref:`selective_reuse` when you run your playbook.
 
-Adding ``tags:`` to a play, or to statically imported tasks and roles, adds
-those tags to all of the contained tasks. This is referred to as *tag
-inheritance*. Tag inheritance is *not* applicable to dynamic inclusions
-such as ``include_role`` and ``include_tasks``.
+Generally speaking, if you want tag inheritance, you probably want to use imports. However, using both includes and imports in a single playbook can lead to difficult-to-diagnose bugs. For this reason, if your playbook uses includes to re-use roles or tasks, and you need tag inheritance on one include, Ansible offers two workarounds. You can use the ``apply`` keyword::
```

Comment:
I would remove the phrase "generally speaking' since it's too much of a colloquialism.